sdk/v3/authn/sessions.go

```diff
@@ -55,23 +55,39 @@ func (t ThirdPartyAuthDetails) MarshalJSON() ([]byte, error) {
 	)
 }
 
+// RootSessionCreateOptions represents useful, optional settings for the
+// creation of a root Session. It currently has no fields, but exists to
+// preserve the possibility of future expansion without having to change client
+// function signatures.
+type RootSessionCreateOptions struct{}
+
+// SessionDeleteOptions represents useful, optional settings for the deletion of
+// a Session. It currently has no fields, but exists to preserve the possibility
+// of future expansion without having to change client function signatures.
+type SessionDeleteOptions struct{}
+
 // SessionsClient is the specialized client for managing Brigade API Sessions.
 type SessionsClient interface {
 	// CreateRootSession creates a Session for the root user (if enabled by the
 	// system administrator) and returns a Token with a short expiry period
 	// (determined by a system administrator). In contrast to most other
 	// operations exposed by the Brigade API, a valid token is not required to
 	// invoke this.
-	CreateRootSession(ctx context.Context, password string) (Token, error)
+	CreateRootSession(
+		ctx context.Context,
+		password string,
+		opts *RootSessionCreateOptions,
+	) (Token, error)
 	// CreateUserSession creates a new User Session and initiates an
 	// authentication workflow with a third-party identity provider. It returns
 	// ThirdPartyAuthDetails containing all information required to continue the
 	// authentication process.
 	CreateUserSession(
-		context.Context, *ThirdPartyAuthOptions,
+		context.Context,
+		*ThirdPartyAuthOptions,
```

Comment:
Good thought. I had considered that at one point and decided against it for reasons I cannot recall now.

There are two different operations for creating a "user session" vs creating a "root session." So I think I would probably opt against `SessionCreateOptions` just because it's ambiguous and over time the options that are applicable to establishing a "user session" vs a "root session" may turn out to be very different from one another.

I could get on board with `UserSessionCreateOptions` and `RootSessionCreateOptions`, respectively. And to your point, this would improve overall naming consistency.

Consider it done.